### PR TITLE
Fixed confusing public and private key requirements and updated gradle.

### LINKED ISSUE #41: dynamically generate key pairs instead of requiring one to be already setup
This improvement will do the following:

1. Eliminate the need to explicitly create a key using the AWS control panel or the aws CLI tool
2. Ask for the AWS key and secret up front.  
3. Move user profile information from a profile_name.yaml into a directory, `profile_name/settings.yaml"
4. Create a key in AWS and save the private key to the user profile.  Use a randomly generated unique name, such as `keyName: "tlp-cluster-368b8370-e148-4590-85b0-573ffaf1ca45"` to avoid conflicts.  Future work may add more identifiers.
5. Remove the question from the startup process asking for the key name & private key location
6. Update documentation accordingly.

### LINKED ISSUE #51: update gradle
gradle version is 4.10.2, latest is 5.2.

```
$ ./gradlew wrapper --gradle-version=5.2.1 --distribution-type=bin
```

### PR DESCRIPTION
**Pull request by rustyrazorblade**
added aws sdk and cleaned up some old cassandra git code
updated gradle to 5.2
Cleaned up logging dependencies and fix profile directory structure.
removed unnecessary log4j dependency
cleanup of the build file
adding the log4j dependency to hijack the SLF4j calls
switch directory structure to work better with multiple profiles
updated documentation

Closes #51
Closes #41

### CODE REVIEW

File: manual/logs/debug.log [hunk 1]
```diff
@@ -0,0 +1,22 @@
+2019-03-16 14:06:18,846 DEBUG c.t.t.MainKt [main] Adding command: init
```
Comment (ossarga) [inline manual/logs/debug.log]: Should this log file be part of the change?

File: src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt [hunk 2]
```diff
@@ -23,14 +39,53 @@ data class User(
             println("We just need to know a few things before we get started.")
 
             val email = Utils.prompt("What's your email?", "")
-            val region = Utils.prompt("What AWS region do you prefer?", "us-west-2")
-            val securityGroup = Utils.prompt("What security group can we put our instances in?  (Must already exist.)", "")
-            val keyName = Utils.prompt("What AWS SSH key pair do you use?", "")
-            val sshKeyPath = Utils.resolveSshKeyPath(Utils.prompt("What is the path to the private key associated with your AWS SSH key pair?", default = ""))
+
+            // we're not honoring it, so we'll take this out
+            //val region = Utils.prompt("What AWS region do you prefer?", "us-west-2")
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt]: We should probably remove this commented out code.

File: src/main/kotlin/com/thelastpickle/tlpcluster/Context.kt [hunk 3]
```diff
@@ -12,11 +12,18 @@ import com.thelastpickle.tlpcluster.configuration.User
 import java.io.File
 import java.nio.file.Files
 
-data class Context(val tlpclusterUserDirectory: File,
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/Context.kt]: This is exactly what I had in mind! 👍

File: src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt [hunk 4]
```diff
@@ -23,14 +39,53 @@ data class User(
...
+            val keyName = "tlp-cluster-${UUID.randomUUID()}"
+            val request = CreateKeyPairRequest.builder()
+                    .keyName(keyName).build()
+
+            val response = ec2.createKeyPair(request)
+
+            // write the private key into the ~/.tlp-cluster/profiles/<profile>/ dir
+
+            val secret = File(context.profileDir, "secret.pem")
+            secret.writeText(response.keyMaterial())
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt]: Two questions here:
1. What permissions does this key have when it is written out? It should be `600`
1. Does the _sshConfig_ reference this key? I was unable to find code that uses it.

File: src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt (hunk 2 above)
Comment (rustyrazorblade) [reply src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt]: I intend on adding it back almost immediately, mind if I leave it?

File: src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt (hunk 4 above)
Comment (rustyrazorblade) [reply src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt]: 1. I just pushed up a change to set the permissions to 600, here's what my secret looks like:

`.rw-------  1.7k     24 jhaddad 19 Mar 12:16  secret.pem`

2. Yep, that logic didn't change.  You'll see an `IdentityFile` line at the top pointing to it after you run `up`

File: src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt (hunk 2 above)
Comment (ossarga) [reply src/main/kotlin/com/thelastpickle/tlpcluster/configuration/User.kt]: That's fine. Sounds good.

**Pull request merged**